### DISCUSSION TITLE
How to do a transitive join?

### DISCUSSION THREAD

**Discussion by dufferzafar** [atoti/atoti#880] [General]
I have three tables: `positions`, `venues`, `prices` with the following schema:

```
positions: 

PK - string - team
PK - string - account
PK - int - id
string - security
long - qty

venues:

PK - string - team
PK - string - account
string - venue

prices:

PK - string - venue
PK - long - id
string - security
double - bid
double - ask
```

I want to join the positions table with the prices table, but notice how the positions table has no `venue` column - but the venues table does!

This is the join I have right now:
```python
positions = session.tables.get("positions")
venues = session.tables.get("venues")
prices = session.tables.get("prices")

# This works well!
positions.join(venues, (positions["team"] == venues["team"]) & (positions["account"] == venues["account"]))
```

I now want to use the venue from the above join, to join positions with price:

```python
positions.join(prices, (positions["venues.venue"] == prices["venue"]) & (positions["id"] == prices["id"]) )
```

This gives me a `KeyError`: 'venues.venue' since positions table doesn't have that column.

How can I access the column venue from the venues table via the positions table since I've already made that join.

Or is there some other way of making this transitive join?

**Comment by OPeyrusse** [comment]
This is not possible at the time. The only solution is to duplicate the venue data in your position table.
